object-filter/src/main/java/org/infinispan/objectfilter/impl/hql/predicate/FilterConjunctionPredicate.java

```diff
@@ -16,6 +16,7 @@ public FilterConjunctionPredicate() {
    @Override
    public BooleanExpr getQuery() {
       // we always expect children.size() >= 1
+      assert children.size() >= 1;
```

Comment:
That assert was only meant for documentation. I think I'll turn it into a comment.